endpoints/openrtb2/auction.go

```diff
@@ -1265,21 +1289,34 @@ func writeError(errs []error, w http.ResponseWriter, labels *pbsmetrics.Labels)
 	return rc
 }
 
-// Returns the effective publisher ID
-func effectivePubID(pub *openrtb.Publisher) string {
-	if pub != nil {
-		if pub.Ext != nil {
-			var pubExt openrtb_ext.ExtPublisher
-			err := json.Unmarshal(pub.Ext, &pubExt)
-			if err == nil && pubExt.Prebid != nil && pubExt.Prebid.ParentAccount != nil && *pubExt.Prebid.ParentAccount != "" {
-				return *pubExt.Prebid.ParentAccount
-			}
+// Sets the effective publisher ID
+func setEffectivePubID(req *openrtb.BidRequest) {
+	var pub *openrtb.Publisher
+	if req.App != nil {
+		if req.App.Publisher == nil {
+			req.App.Publisher = new(openrtb.Publisher)
 		}
-		if pub.ID != "" {
-			return pub.ID
+		pub = req.App.Publisher
+	}
+	if req.Site != nil {
+		if req.Site.Publisher == nil {
+			req.Site.Publisher = new(openrtb.Publisher)
 		}
+		pub = req.Site.Publisher
+	}
+
+	if pub.Ext != nil {
+		var pubExt openrtb_ext.ExtPublisher
+		err := json.Unmarshal(pub.Ext, &pubExt)
+		if err == nil && pubExt.Prebid != nil && pubExt.Prebid.ParentAccount != nil &&
+			*pubExt.Prebid.ParentAccount != "" {
+			pub.ID = *pubExt.Prebid.ParentAccount
```

Comment:
Why are you setting the publisher ID to the parent account? This is a breaking change in behavior. Parent account was created to handle the case where the account in the PBS context does not match the publisher ID agreed upon in the publisher-bidder context.